### PR TITLE
build(topology): add backward compatibility check for protobuf messages

### LINKED ISSUE #15743: Add backward compatibility check for protobuf definitions in topology module
Similar to gatway grpc protocol, we can use the `proto-backwards-compatibility` plugin.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

Add maven plugin to prevent backward incompatible changes to protobuf message in topology module.

## Related issues

closes #15743

### CODE REVIEW

**Comment by deepthidevaki** [comment]
I think this should be also backported to release 8.4, to ensure future patch versions are not breaking compatability.

File: topology/src/main/resources/proto/proto.lock [hunk 1]
```diff
@@ -0,0 +1,549 @@
+{
+  "definitions": [
+    {
+      "protopath": "requests.proto",
```
Comment (oleschoenburg) [inline topology/src/main/resources/proto/proto.lock:4]: :thought_balloon: Do we want to / need to keep backwards compatibility here? The requests are ephemeral and it should be fine for scaling not to work during rolling updates, right?

**Review by lenaschoenburg — APPROVED** [review]
:+1:

File: topology/src/main/resources/proto/proto.lock (hunk 1 above)
Comment (deepthidevaki) [reply topology/src/main/resources/proto/proto.lock:4]: By default, let's try not to keep the compatibility. When we have to break compatibility, we can do that deliberately.

**Comment by deepthidevaki** [comment]
bors merge

**Pull request merged**